File: src/main/java/com/adjorno/billib/rest/model/TrendList.java
package com.adjorno.billib.rest.model;

import com.adjorno.billib.rest.db.Track;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class TrendList {
    @SerializedName("name")
    private String mTrendName;

    @SerializedName("tracks")
    private List<Track> mTracks = new ArrayList<>();

    public TrendList(String trendName) {
        mTrendName = trendName;
    }

    public String getTrendName() {
        return mTrendName;
    }

    public void setTrendName(String trendName) {
        mTrendName = trendName;
    }

    public List<Track> getTracks() {
        return mTracks;
    }

    public void addTrack(Track track) {
        mTracks.add(track);
    }
}

File: backend/src/main/java/com/ifochka/m14n/rest/trends/rest/dtos/TrendList.kt
package com.ifochka.m14n.rest.trends.rest.dtos

import com.ifochka.m14n.rest.catalog.domain.Track
import java.util.ArrayList

data class TrendList(
    val name: String,
) {
    val tracks: MutableList<Track> = ArrayList()
}
